tfhe-benchmark/benches/core_crypto/ks_bench.rs

```diff
@@ -588,157 +591,199 @@ mod cuda {
+                                })
+                            });
+                        }
                     }
                 }
                 BenchmarkType::Throughput => {
                     let gpu_keys_vec = cuda_local_keys_core(&cpu_keys, None);
                     let gpu_count = get_number_of_gpus() as usize;
 
                     bench_id = format!("{bench_name}::throughput::{name}");
-                    let blocks: usize = 1;
-                    let elements = throughput_num_threads(blocks, 1);
-                    let elements_per_stream = elements as usize / gpu_count;
-                    bench_group.throughput(Throughput::Elements(elements));
-                    bench_group.sample_size(50);
-                    bench_group.bench_function(&bench_id, |b| {
-                        let setup_encrypted_values = || {
-                            let local_streams = cuda_local_streams_core();
-
-                            let plaintext_list = PlaintextList::new(
-                                Scalar::ZERO,
-                                PlaintextCount(elements_per_stream),
-                            );
 
-                            let input_lwe_lists = (0..gpu_count)
-                                .map(|i| {
-                                    let mut input_ct_list = LweCiphertextList::new(
-                                        Scalar::ZERO,
-                                        lwe_sk.lwe_dimension().to_lwe_size(),
-                                        LweCiphertextCount(glwe_sk.polynomial_size().0),
-                                        ciphertext_modulus,
-                                    );
-                                    encrypt_lwe_ciphertext_list(
-                                        &lwe_sk,
-                                        &mut input_ct_list,
-                                        &plaintext_list,
-                                        params.lwe_noise_distribution.unwrap(),
-                                        &mut encryption_generator,
-                                    );
+                    let local_streams = cuda_local_streams_core();
+                    let mem_size = get_packing_keyswitch_list_64_size_on_gpu(
+                        &local_streams[0],
+                        lwe_sk.lwe_dimension(),
+                        glwe_sk.glwe_dimension(),
+                        glwe_sk.polynomial_size(),
+                        LweCiphertextCount(glwe_sk.polynomial_size().0),
+                    );
 
-                                    CudaLweCiphertextList::from_lwe_ciphertext_list(
-                                        &input_ct_list,
-                                        &local_streams[i],
-                                    )
-                                })
-                                .collect::<Vec<_>>();
+                    skip_bench = false;
+                    for gpu_index in 0..gpu_count {
```

Comment:
Here you only need to check on gpu_indexes[0], which is where the function is executed.